### ISSUE TITLE
数据集和尺寸的问题

### ISSUE THREAD

**Issue by lulu732** [Caoxuheng/HIFtool#17]
大佬您好，您代码中的数据集是怎么制作的呢，pavia数据集中的波段数与网站中下载的原始数据不同。此外，代码中如何实现图像裁剪，使用原尺寸容易导致爆显存。 十分感谢您的分享。

**Comment by Caoxuheng** [comment]
Pavia的数据已经把元素吸收波段和噪音波段的红外部分去除了。
PaviaDataset中的patch_h=3*1, patch_w=3*1,调整尺寸，这里是LR-HSI的尺寸。对应的HR-HSI默认为3*32 3*32

**Comment by lulu732** [comment]
使用MSSTNET训练pavia，训练验证中psnr可以达到54左右，但是测试却只有30出头，这样合理吗。

    Train_data = PaviaDataset(f'Multispectral Image Dataset\{dataset_name}\{dataset_name}.mat',type='train')
    Val_data =  PaviaDataset(f'Multispectral Image Dataset\{dataset_name}\{dataset_name}.mat',type='eval')
   Test_data = PaviaDataset('Multispectral Image Dataset/Pavia/GT.mat', type='test')

Loader中，只修改了   
def getData_(self):
        srf = sio.loadmat('Dataloader_tool/srflib/chikuseisrf.mat')['R']
        hrhsi = sio.loadmat(self.mat_save_path )['HSI']
其他没有修改







































































M

**Comment by lulu732** [comment]
PSRT模型，在ratio＝16训练后，验证psnr=45，用GT.mat测试得到psnr=42.95。ratio=8训练，验证psnr=44，用GT.mat测试得到psnr=23.7338。模型是在循环到第50次左右训练psnr基本稳定的，后续只有小幅度的提升。
if __name__=='__main__':

    from torch.utils.data import DataLoader
    from Dataloader_tool import PaviaDataset
    # Build Network
    Method = 'PSRT'

    model, opt = model_generator(Method,'cuda')
    # Dataset Setting
    dataset_name = 'Pavia'
    model_folder = 'PretrainModel/' + Method + '/'+ dataset_name +'/'

    Batch_size = 1
    start = 1000
    Test_data = PaviaDataset('Multispectral Image Dataset/Pavia/GT.mat', type='test')

    training_data_loader = DataLoader(dataset=Test_data, num_workers=0, batch_size=Batch_size, shuffle=False,
                                      pin_memory=True, drop_last=False)
    output_HRHSI = Eva(model,training_data_loader,model_folder=model_folder,start_epoch=start,name=Method)
这段测试了start=50，100，200，500，1000，结果都远低于验证时的psnr。

**Comment by Caoxuheng** [comment]
我看一下，你是8倍训练，这个代码适配的都是32倍的。

---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2025年05月08日 11:21 |
| 收件人 | Caoxuheng/HIFtool ***@***.***> |
| 抄送人 | Xuheng ***@***.***>,
Comment ***@***.***> |
| 主题 | Re: [Caoxuheng/HIFtool] 数据集和尺寸的问题 (Issue #17) |
lulu732 left a comment (Caoxuheng/HIFtool#17)

PSRT模型，在ratio＝16训练后，验证psnr=45，用GT.mat测试得到psnr=42.95。ratio=8训练，验证psnr=44，用GT.mat测试得到psnr=23.7338。模型是在循环到第50次左右训练psnr基本稳定的，后续只有小幅度的提升。
if name=='main':

from torch.utils.data import DataLoader
from Dataloader_tool import PaviaDataset
# Build Network
Method = 'PSRT'

model, opt = model_generator(Method,'cuda')
# Dataset Setting
dataset_name = 'Pavia'
model_folder = 'PretrainModel/' + Method + '/'+ dataset_name +'/'

Batch_size = 1
start = 1000
Test_data = PaviaDataset('Multispectral Image Dataset/Pavia/GT.mat', type='test')

training_data_loader = DataLoader(dataset=Test_data, num_workers=0, batch_size=Batch_size, shuffle=False,
                                  pin_memory=True, drop_last=False)
output_HRHSI = Eva(model,training_data_loader,model_folder=model_folder,start_epoch=start,name=Method)


这段测试了start=50，100，200，500，1000，结果都远低于验证时的psnr。

**Comment by lulu732** [comment]
您测试出来了吗，是过拟合的缘故吗。

**Comment by Caoxuheng** [comment]
可能是测试代码的问题，最近事务比较多，调试好了会更新一下。

---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2025年05月13日 15:26 |
| 收件人 | Caoxuheng/HIFtool ***@***.***> |
| 抄送人 | Xuheng ***@***.***>,
Comment ***@***.***> |
| 主题 | Re: [Caoxuheng/HIFtool] 数据集和尺寸的问题 (Issue #17) |
lulu732 left a comment (Caoxuheng/HIFtool#17)

您测试出来了吗，是过拟合的缘故吗。

**Comment by lulu732** [comment]
使用xiongan数据集，发现DCTransformer训练非常缓慢，16倍几乎是一小时只跑两轮epoch，其他模型也很慢，这是xiongan数据集的特性吗？

**Comment by Caoxuheng** [comment]
是的，雄安数据集比较大。尝试增大batchsize。

---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2025年05月21日 14:20 |
| 收件人 | Caoxuheng/HIFtool ***@***.***> |
| 抄送人 | Xuheng ***@***.***>,
Comment ***@***.***> |
| 主题 | Re: [Caoxuheng/HIFtool] 数据集和尺寸的问题 (Issue #17) |
lulu732 left a comment (Caoxuheng/HIFtool#17)

使用xiongan数据集，发现DCTransformer训练非常缓慢，16倍几乎是一小时只跑两轮epoch，其他模型也很慢，这是xiongan数据集的特性吗？

**Comment by lulu732** [comment]
您好，使用HySure融合得到的图像，为什么都出现了很明显的网格

![Image](https://github.com/user-attachments/assets/5a23a6df-2e7b-47b0-af32-2403f1471424)

**Comment by Caoxuheng** [comment]
输入HySure的空间退化模型有误差。HySure子函数 equip()中的PSF 要和生成数据集中的退化核一致才可以。

**Comment by Caoxuheng** [comment]
输入HySure的空间退化模型有误差。HySure子函数 equip()中的PSF 要和生成数据集中的退化核一致才可以。
